### PR TITLE
[ONNX][dynamo_export] Add 'aten::rsub' type promotion

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113780
* #113703
* __->__ #113697
* #113404

The logic is the same as 'aten::sub'. Needed by llama2.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

File: test/onnx/test_fx_op_consistency.py [hunk 1]
```diff
@@ -145,11 +146,12 @@
         "nn.functional.normalize",
         # "nn.functional.scaled_dot_product_attention"  non-deterministic
         "nonzero",
-        "linalg.vector_norm",
```
Comment (thiagocrepaldi) [inline test/onnx/test_fx_op_consistency.py:148]: thanks, I guess my string sorting skills failed me lol

**Review by thiagocrepaldi — APPROVED** [review]

**Comment by BowenBao** [comment]
@huydhn looks like it is some float16 test with precision issue, let me retry with relaxing the constraint.

**Pull request closed without merging**